### DISCUSSION TITLE
Selecting children/siblings/ancestors/descendants of a specific node

### DISCUSSION THREAD

**Discussion by rob-gordon** [cytoscape/cytoscape.js#3080] [General]
Does anyone know if it is possible to use [cytoscape selectors](https://js.cytoscape.org/#selectors) to select nodes which are children, siblings, ancestors, or descendants of a particular node?

### Example 1
A selector which selects all nodes which are one edge away from a given node
<img width="398" alt="Screen Shot 2022-11-28 at 9 24 14 AM" src="https://user-images.githubusercontent.com/159949/204301649-098dd64e-0ebc-48f6-bea6-20d169561693.png">

An example selector, `#a:sibling { background-color: yellow; }`
<img width="394" alt="Screen Shot 2022-11-28 at 9 24 22 AM" src="https://user-images.githubusercontent.com/159949/204301696-bf6626b9-aaa9-41cb-b266-d3268f32900a.png">

### Example 2
A selector which selects all the descendants of a node in a directed acyclic graph
<img width="167" alt="Screen Shot 2022-11-28 at 9 21 34 AM" src="https://user-images.githubusercontent.com/159949/204301075-2d02b082-ec56-4134-83f0-e07429b85220.png">

An example selector, `#d:descendants { background-color: yellow; }`
<img width="173" alt="Screen Shot 2022-11-28 at 9 21 41 AM" src="https://user-images.githubusercontent.com/159949/204301098-2f44b6ef-e182-4dbb-b809-336b5874043c.png">

I realize these could be expensive to compute and probably ambiguous in some contexts (I'm not a graph theory expert), but does anyone know if there has been an attempt to support selectors like these that I could try it out? Alternatively, is this functionality that could be added via a plugin?

Thanks for any advice or thoughts.

**Comment by maxkfranz** [comment]
Topology is too expensive to be pragmatic for selectors.  Cytoscape has a number of functions related to traversal that you can use: https://js.cytoscape.org/#collection/traversing

**Reply by rob-gordon** [reply]
That's what I figured. If it were easy and inexpensive it would probably already be in Cy. Okay, I'll find another way to accomplish styling these using the traversal functions. Thanks for your help!
